app/controllers/foreman_rh_cloud/concerns/api/v2/hosts_controller_extensions.rb

```diff
@@ -0,0 +1,17 @@
+module ForemanRhCloud
+  module Concerns
+    module Api::V2::HostsControllerExtensions
+      extend ActiveSupport::Concern
+
+      included do
+        def find_resource
+          insights_host = InsightsFacet.find_by(uuid: params[:id])&.host
+          Rails.logger.debug "ForemanRhCloud host: #{@host}"
+          params[:id] = insights_host.name if insights_host.present?
+          @host = insights_host || super
```

Comment:
I'm still getting:
```
Couldn't find Host::Managed with 'id'=2a7bfadc-453a-4ef0-a67e-390acafa966a [WHERE "hosts"."type" = $1 AND "hosts"."organization_id" = $2 AND "hosts"."location_id" = $3]
```
on API call to `/api/hosts/2a7bfadc-453a-4ef0-a67e-390acafa966a?show_hidden_parameters=true`

Reply:
I extended `/api/v2/hosts` and `/hosts` but not `/api/hosts`. Do I need to extend all three?

Reply:
update: it appears there's no `::Api::HostsController` to extend.